subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/FlatDirResolveIntegrationTest.groovy

```diff
@@ -31,7 +31,7 @@ configurations { compile }
 dependencies {
     compile 'group:a:1.4'
     compile 'group:b:2.0'
-    compile 'group:c:'
+    compile 'group:c:0.1'
```

Comment:
❌ This is a breaking change. Flat dir repos don't need a version. I.e. you can just have `guava.jar`.

Reply:
We probably need to do this deeper down in the individual repo implementations. I already started this for Maven, because I had seen it as a performance problem for a customer. (Though there it was a missing groupID, not version). https://github.com/gradle/gradle/pull/3431

Reply:
If we want to keep this use case working, I think you are right.

Then we should reject this PR and, go with #3431 and do the same for ivy repositories.

Reply:
#3431 also solves the issue that we query without having a version in the context of BOM support (#3383).

https://github.com/gradle/gradle/pull/3383/files#diff-fa64843d7d8a6140d43892539f45e50aR29

Reply:
Let's go with #3431 then and do the same for Ivy.